include/exiv2/error.hpp

```diff
@@ -260,7 +260,7 @@ namespace Exiv2 {
              provided to print errors to a stream.
      */
     template<typename charT>
-    class BasicError : public AnyError {
+    class EXIV2API BasicError : public AnyError {
```

Comment:
This breaks builds on Windows and probably opens quite the can of worms: declaring a template as export/visible is not really supported.

Reply:
Right ... the last time I worked on this topic we reached the same conclusion ... but somehow we always forget about it when somebody opens an issue in github related with that part of the code. 

I guess the final fix for this will be to finish the work started in #685

Reply:
Oh gosh, this isn't good news.  The changes in #685 are huge.  Because they will change the API, I'd prefer not to do that in 0.27-maintenance.

I don't think we're happy with the following hack:
```
#ifdef __FREEBSD___
class EXIV2API BasicError : public AnyError
else
class BasicError : public AnyError
#endif
```
And we should do something similar on the change in src/CMakeLists.txt

For sure, we're not putting this into v0.27.2.  We need to explore other possibilities.  @lantw44 can you comment on this please?

Reply:
Does it work if we drop `EXIV2API` from `setMsg` in include/exiv2/error.hpp, and move "free functions, template and inline definitions" to src/error.cpp or mark them as `inline`?

Reply:
Useful suggestions, thanks.  I'll wait to see if other in Team Exiv2 wish to comment and I'll investigate this evening.

I've created a fresh FreeBSD VM this morning to investigate your thoughts about not using the package manager to install GCC.

Reply:
> Does it work if we drop `EXIV2API` from `setMsg` in include/exiv2/error.hpp, and move "free functions, template and inline definitions" to src/error.cpp or mark them as `inline`?

`setMsg` must stay in the header because it is used in the constructor of `BasicError` and because of that it must be marked as `EXIV2API` as it otherwise will not be visible on Windows. Marking it inline is worth a try though, as that's probably the only option beside #685.